File: backend/app/routers/environment.py
from fastapi import APIRouter, Depends

from app.core.auth import get_org_id
from app.schemas.environment import HealthReport
from app.services.environment_scanner import scan_environment

router = APIRouter(prefix="/api/v1/environment", tags=["environment"])


@router.get("/scan", response_model=HealthReport)
async def scan_env(org_id: str = Depends(get_org_id)) -> HealthReport:
    """Scan the environment and return health report."""
    _ = org_id
    return scan_environment()

File: app/src/routers/environment.js
import { Router } from "express";

import { requireOrgId } from "../core/auth.js";
import { createError } from "../core/errors.js";
import { parseEnvironmentScanRequest } from "../schemas/environment.js";
import { scanEnvironment } from "../services/environmentService.js";

const router = Router();

router.post("/scan", requireOrgId, async (req, res, next) => {
  try {
    const payload = parseEnvironmentScanRequest(req.body);
    const report = await scanEnvironment(req.orgId, payload.repoPath);
    res.json(report);
  } catch (err) {
    if (err instanceof Error && err.message.includes("Repository path")) {
      next(createError(400, "invalid_request", err.message));
      return;
    }
    next(err);
  }
});

router.get("/health", requireOrgId, async (req, res, next) => {
  try {
    const repoPath = typeof req.query.repo_path === "string"
      ? req.query.repo_path
      : typeof req.query.repoPath === "string"
        ? req.query.repoPath
        : null;
    const report = await scanEnvironment(req.orgId, repoPath);
    res.json(report);
  } catch (err) {
    if (err instanceof Error && err.message.includes("Repository path")) {
      next(createError(400, "invalid_request", err.message));
      return;
    }
    next(err);
  }
});

export default router;